[WPE] WPE Platform: add monitors API

carlosgcampos opened the pull request:
#### fda6216679915707208bd7794ae22c0c92b808cd
<pre>
[WPE] WPE Platform: add monitors API
<a href="https://bugs.webkit.org/show_bug.cgi?id=265639">https://bugs.webkit.org/show_bug.cgi?id=265639</a>

Reviewed by Adrian Perez de Castro.

Add monitors API with implementation for the wayland platform for now.
This API is used inside WebKit to set the screen properties, set the
platform display ID and get the refresh rate for display link.

* Source/WebCore/platform/PlatformScreen.cpp:
* Source/WebCore/platform/ScreenProperties.h:
* Source/WebCore/platform/wpe/PlatformScreenWPE.cpp:
(WebCore::widgetDisplayID):
(WebCore::screenDepth):
(WebCore::screenDepthPerComponent):
(WebCore::screenIsMonochrome):
(WebCore::screenDPI):
(WebCore::screenRect):
(WebCore::screenAvailableRect):
(WebCore::setScreenDPIObserverHandler): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.serialization.in:
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::m_backend):
(WKWPE::View::~View):
(WKWPE::View::updateDisplayID):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
(WebKit::findCrtc):
(WebKit::DisplayVBlankMonitorDRM::create):
* Source/WebKit/UIProcess/glib/ScreenManager.cpp: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
(WebKit::ScreenManager::singleton):
(WebKit::ScreenManager::displayID const):
(WebKit::ScreenManager::monitor const):
(WebKit::ScreenManager::addMonitor):
(WebKit::ScreenManager::removeMonitor):
(WebKit::ScreenManager::propertiesDidChange const):
* Source/WebKit/UIProcess/glib/ScreenManager.h: Renamed from Source/WebKit/UIProcess/gtk/ScreenManager.h.
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/gtk/ScreenManagerGtk.cpp: Renamed from Source/WebKit/UIProcess/gtk/ScreenManager.cpp.
(WebKit::ScreenManager::generatePlatformDisplayID):
(WebKit::ScreenManager::ScreenManager):
(WebKit::ScreenManager::collectScreenProperties const):
* Source/WebKit/UIProcess/wpe/ScreenManagerWPE.cpp: Added.
(WebKit::ScreenManager::generatePlatformDisplayID):
(WebKit::ScreenManager::ScreenManager):
(WebKit::ScreenManager::collectScreenProperties const):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpe_display_class_init):
(wpe_display_get_n_monitors):
(wpe_display_get_monitor):
(wpe_display_monitor_added):
(wpe_display_monitor_removed):
(wpeDisplayDispose): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.h:
* Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp: Added.
(wpeMonitorGetProperty):
(wpe_monitor_class_init):
(wpe_monitor_get_id):
(wpe_monitor_invalidate):
(wpe_monitor_get_x):
(wpe_monitor_get_y):
(wpe_monitor_set_position):
(wpe_monitor_get_width):
(wpe_monitor_get_height):
(wpe_monitor_set_size):
(wpe_monitor_get_physical_width):
(wpe_monitor_get_physical_height):
(wpe_monitor_set_physical_size):
(wpe_monitor_get_scale):
(wpe_monitor_set_scale):
(wpe_monitor_get_refresh_rate):
(wpe_monitor_set_refresh_rate):
* Source/WebKit/WPEPlatform/wpe/WPEMonitor.h: Added.
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpeViewGetProperty):
(wpe_view_class_init):
(wpe_view_get_monitor):
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/wayland/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandDispose):
(wpeDisplayWaylandGetNMonitors):
(wpeDisplayWaylandGetMonitor):
(wpeDisplayWaylandFindMonitor):
(wpe_display_wayland_class_init):
(wpeDisplayWaylandGetOutput): Deleted.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWaylandPrivate.h:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEMonitorWayland.cpp: Added.
(wpeMonitorWaylandInvalidate):
(wpeMonitorWaylandDispose):
(wpe_monitor_wayland_class_init):
(wpe_monitor_wayland_get_wl_output):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEMonitorWayland.h: Copied from Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(wpeViewWaylandConstructed):
(wpeViewWaylandDispose):
(wpeViewWaylandGetMonitor):
(wpe_view_wayland_class_init):
* Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h:
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):

Canonical link: <a href="https://commits.webkit.org/274090@main">https://commits.webkit.org/274090@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29ca24ddca35e9771bb6c569aac85912e74ae5f1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37126 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16032 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39426 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39703 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33148 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38419 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18600 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13169 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31654 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37689 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13530 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32694 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11789 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11790 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33277 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40959 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33630 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33809 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37690 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12108 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9904 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35831 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13740 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12471 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4912 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13004 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebCore/platform/PlatformScreen.cpp:
```diff
@@ -26,7 +26,7 @@
 #include "PlatformScreen.h"
 
-#if PLATFORM(COCOA) || PLATFORM(GTK)
+#if PLATFORM(COCOA) || PLATFORM(GTK) || (PLATFORM(WPE) && ENABLE(WPE_PLATFORM))
```
We could make the guard be only `ENABLE(WPE_PLATFORM)` because that implies that `ENABLE(WPE)` is also enabled:

```suggestion
#if PLATFORM(COCOA) || PLATFORM(GTK) || ENABLE(WPE_PLATFORM)
```

carlosgcampos replied:
Yes, I thought about this, but I found a bit weird to use an ENABLE macro specific to a platform without the explicit platform check. The WPE_PLATFORM macro will be removed eventually and only the platform check would be used.

aperezdc replied:
Fair point 👌🏼

aperezdc commented on Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
```diff
@@ -52,15 +55,20 @@
+#if PLATFORM(GTK) || (PLATFORM(WPE) && ENABLE(WPE_PLATFORM))
+static std::optional<std::pair<uint32_t, uint32_t>> findCrtc(int fd, PlatformMonitor* monitor)
 {
     drmModeRes* resources = drmModeGetResources(fd);
```
Unrelated to the content of the patch itself... At some point we may want to have `std::unique_ptr` specializations for libdrm types, so we can use RAII with them and for example in this function would could then return the value directly and let the compiler handle freeing resources for us.

carlosgcampos replied:
We have them in the wpe-platform indeed, but they can't be shared, so we would need to duplicate in WebKit layer.

aperezdc commented on Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
```diff
@@ -176,9 +191,32 @@ static int crtcBitmaskForIndex(uint32_t crtcIndex)
 std::unique_ptr<DisplayVBlankMonitor> DisplayVBlankMonitorDRM::create(PlatformDisplayID displayID)
 {
+#if PLATFORM(WPE) && ENABLE(WPE_PLATFORM)
+    bool usingWPEPlatformAPI = !!g_type_class_peek(WPE_TYPE_DISPLAY);
```
This could be `static`.

```suggestion
    static bool usingWPEPlatformAPI = !!g_type_class_peek(WPE_TYPE_DISPLAY);
```

aperezdc commented on Source/WebKit/UIProcess/glib/ScreenManager.h:
```diff
@@ -43,20 +51,24 @@ class ScreenManager {
-    Vector<GRefPtr<GdkMonitor>, 1> m_monitors;
-    HashMap<GdkMonitor*, PlatformDisplayID> m_monitorToDisplayIDMap;
+    Vector<GRefPtr<PlatformMonitor>, 1> m_monitors;
+    HashMap<PlatformMonitor*, PlatformDisplayID> m_monitorToDisplayIDMap;
```
I know it's out of the scope for this patch, but I'll comment on this anyway: How many monitors are we *realistically* going to have? Typically one, usually two as maximum, in some setups may go up to 3 or 4.

It would be actually better to iterate a few elements in a vector doing a linear search than going through hashing due to better cache locality and memory prefetching. At some point we may want to change this to remove the `HashMap` in a different patch 😉

carlosgcampos replied:
I see your point, maybe we can just do it now.

carlosgcampos replied:
We need the map, because in the case of GTK, the id is generated and not returned by the monitor, so this map is the only reference we have between a GdkMonitor and its generated id. So, we could do it for WPE, but better keep this and share the code, I think

aperezdc replied:
Ah, I see. Let's then keep it and share the code.

aperezdc commented on Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
```diff
@@ -158,8 +161,16 @@ void WebProcessPool::platformInitializeWebProcess(const WebProcessProxy& process
     parameters.gtkSettings = GtkSettingsManager::singleton().settingsState();
+#endif
+
+#if PLATFORM(GTK)
```
Why not merging this with the previous set of guards?

```suggestion
```

aperezdc commented on Source/WebKit/UIProcess/wpe/ScreenManagerWPE.cpp:
```diff
@@ -0,0 +1,95 @@
+        data.screenDepth = 24;
+        data.screenDepthPerComponent = 8;
+        data.screenSize = { wpe_monitor_get_physical_width(monitor), wpe_monitor_get_physical_height(monitor) };
+        static const double millimetresPerInch = 25.4;
```
This can be `constexpr` to make sure it doesn't end in the read-only data section of the library:

```suggestion
        static constexpr double millimetresPerInch = 25.4;
```

aperezdc commented on Source/WebKit/UIProcess/wpe/ScreenManagerWPE.cpp:
```diff
@@ -0,0 +1,95 @@
+        ScreenData data;
+        data.screenRect = FloatRect(wpe_monitor_get_x(monitor), wpe_monitor_get_y(monitor), width, height);
+        data.screenAvailableRect = data.screenRect;
+        data.screenDepth = 24;
+        data.screenDepthPerComponent = 8;
```
Any chance of un-hardcoding this? It would be fine to do it in a follow-up, but I wonder what your thoughts are in this regard.

carlosgcampos replied:
They are already hardcoded for wpe in PlatformScreen, I just copied it. I don't know if we can get this info.

aperezdc replied:
I reckoned it wouldn't be trivial, that's why I asked if you had any thought about it. Anyway, no problem.

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
```diff
@@ -84,6 +93,38 @@ static void wpe_display_class_init(WPEDisplayClass* displayClass)
+     * @display: a #WPEDisplay
+     * @monitor: the #WPEMonitor removed
+     *
+     * Emitted when a monitor is removed
```
The documentation is ambiguous. We should clarify whether it is emitted _before_ the monitor is removed or emitted _after_ it is removed (where the monitor object itself is still usable but not part of the list of ones in use). From the `wpe_display_monitor_removed()` implementation below I understand it's the latter.

carlosgcampos replied:
Good point!

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp:
```diff
@@ -0,0 +1,520 @@
+
+/**
+ * WPEMonitor:
+ *
```
Missing documentation.

carlosgcampos replied:
This is the case of most (if not all) the new API objects, I plan to fill those once the API is more stable.

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp:
```diff
@@ -0,0 +1,520 @@
+    /**
+     * WPEMonitor:x:
+     *
+     * The x coordinate of the monitor position.
```
Is this in pixels, cm, inches? Seems to be pixels, it would be better to state it explicitly.

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp:
```diff
@@ -0,0 +1,520 @@
+    /**
+     * WPEMonitor:physical-width:
+     *
+     * The physical with of the monitor in millimeters.
```
Typo: `s/with/width` &mdash; good that the physical size is stated to be in millimiters 😉

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp:
```diff
@@ -0,0 +1,520 @@
+ * @monitor: a #WPEMonitor
+ *
+ * Get the @monitor identifier
+ *
```
May the identifier be zero? It seems worth telling in the documentation that it is a non-zero value, and unique for each one monitor instance.

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp:
```diff
@@ -0,0 +1,520 @@
+ * wpe_monitor_invalidate:
+ * @monitor: a #WPEMonitor
+ *
+ * Invalidate @monitor
```
What does “invalidating” a monitor supposed to do? As an user of the API, it would be good to know, and why/when one should use this method.

aperezdc commented on Source/WebKit/WPEPlatform/wpe/wayland/WPEMonitorWayland.cpp:
```diff
@@ -0,0 +1,141 @@
+        wpe_monitor_set_refresh_rate(monitor, refresh);
+    },
+    // done
+    [](void*, struct wl_output*)
+    {
+    },
+    // scale
```
I wonder if it would be desirable to thaw property notifications for the monitor object while the compositor is sending events with the different values, and then unfreeze them in the `.done` callback here. Although maybe it's not an issue that the property updates are notified as updates progress. Not sure, just a thought.

carlosgcampos replied:
We only connect to notify::scale so I don't think it matter much.

aperezdc replied:
Let's leave it as-is for now. We can always revisit it later if needed.

aperezdc reviewed (changes requested):
This looks good to me. There is nothing that should be blocking this from landing, other than a couple of doubts and questions that crossed my mind while reading through this. Thanks @carlosgcampos!

carlosgcampos committed:
[WPE] WPE Platform: add monitors API
https://bugs.webkit.org/show_bug.cgi?id=265639

Reviewed by Adrian Perez de Castro.

Add monitors API with implementation for the wayland platform for now.
This API is used inside WebKit to set the screen properties, set the
platform display ID and get the refresh rate for display link.

* Source/WebCore/platform/PlatformScreen.cpp:
* Source/WebCore/platform/ScreenProperties.h:
* Source/WebCore/platform/wpe/PlatformScreenWPE.cpp:
(WebCore::widgetDisplayID):
(WebCore::screenDepth):
(WebCore::screenDepthPerComponent):
(WebCore::screenIsMonochrome):
(WebCore::screenDPI):
(WebCore::screenRect):
(WebCore::screenAvailableRect):
(WebCore::setScreenDPIObserverHandler): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.serialization.in:
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::m_backend):
(WKWPE::View::~View):
(WKWPE::View::updateDisplayID):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
(WebKit::findCrtc):
(WebKit::DisplayVBlankMonitorDRM::create):
* Source/WebKit/UIProcess/glib/ScreenManager.cpp: Copied from Source/WebKit/UIProcess/gtk/ScreenManager.h.
(WebKit::ScreenManager::singleton):
(WebKit::ScreenManager::displayID const):
(WebKit::ScreenManager::monitor const):
(WebKit::ScreenManager::addMonitor):
(WebKit::ScreenManager::removeMonitor):
(WebKit::ScreenManager::propertiesDidChange const):
* Source/WebKit/UIProcess/glib/ScreenManager.h: Renamed from Source/WebKit/UIProcess/gtk/ScreenManager.h.
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/gtk/ScreenManagerGtk.cpp: Renamed from Source/WebKit/UIProcess/gtk/ScreenManager.cpp.
(WebKit::ScreenManager::generatePlatformDisplayID):
(WebKit::ScreenManager::ScreenManager):
(WebKit::ScreenManager::collectScreenProperties const):
* Source/WebKit/UIProcess/wpe/ScreenManagerWPE.cpp: Added.
(WebKit::ScreenManager::generatePlatformDisplayID):
(WebKit::ScreenManager::ScreenManager):
(WebKit::ScreenManager::collectScreenProperties const):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpe_display_class_init):
(wpe_display_get_n_monitors):
(wpe_display_get_monitor):
(wpe_display_monitor_added):
(wpe_display_monitor_removed):
(wpeDisplayDispose): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.h:
* Source/WebKit/WPEPlatform/wpe/WPEMonitor.cpp: Added.
(wpeMonitorGetProperty):
(wpe_monitor_class_init):
(wpe_monitor_get_id):
(wpe_monitor_invalidate):
(wpe_monitor_get_x):
(wpe_monitor_get_y):
(wpe_monitor_set_position):
(wpe_monitor_get_width):
(wpe_monitor_get_height):
(wpe_monitor_set_size):
(wpe_monitor_get_physical_width):
(wpe_monitor_get_physical_height):
(wpe_monitor_set_physical_size):
(wpe_monitor_get_scale):
(wpe_monitor_set_scale):
(wpe_monitor_get_refresh_rate):
(wpe_monitor_set_refresh_rate):
* Source/WebKit/WPEPlatform/wpe/WPEMonitor.h: Added.
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpeViewGetProperty):
(wpe_view_class_init):
(wpe_view_get_monitor):
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/wayland/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandDispose):
(wpeDisplayWaylandGetNMonitors):
(wpeDisplayWaylandGetMonitor):
(wpeDisplayWaylandFindMonitor):
(wpe_display_wayland_class_init):
(wpeDisplayWaylandGetOutput): Deleted.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWaylandPrivate.h:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEMonitorWayland.cpp: Added.
(wpeMonitorWaylandInvalidate):
(wpeMonitorWaylandDispose):
(wpe_monitor_wayland_class_init):
(wpe_monitor_wayland_get_wl_output):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEMonitorWayland.h: Copied from Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(wpeViewWaylandConstructed):
(wpeViewWaylandDispose):
(wpeViewWaylandGetMonitor):
(wpe_view_wayland_class_init):
* Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h:
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):

Canonical link: https://commits.webkit.org/274090@main
